File: fahim_hasnat/414. Third Maximum Number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var thirdMax = function(nums) {
    nums.sort((a, b) => b - a); // Sort in descending order

    let distinctCount = 1;
    let last = nums[0];

    for (let i = 1; i < nums.length; i++) {
        if (nums[i] !== last) {
            distinctCount++;
            last = nums[i];
        }
        if (distinctCount === 3) {
            return nums[i];
        }
    }

    return nums[0];
};

File: fahim_hasnat/414. Third Maximum Number.go
// https://leetcode.com/problems/third-maximum-number/

func thirdMax(nums []int) int {
    sort.Sort(sort.Reverse(sort.IntSlice(nums)))

    distinctCount := 1
    last := nums[0]

    for _, n := range nums {
        if n != last {
            distinctCount++
            last = n
        }
        if distinctCount == 3 {
            return n
        }
    }

    return nums[0]
}
